### DISCUSSION TITLE
Geoid seperation

### DISCUSSION THREAD

**Discussion by brinpat** [iTwin/community#9] [General]
Hi,

How do I implement geoid separation with value of 1 into my web app?

I believe it may go into the code somewhere here:

```
  /** Makes ascetic changes to the default view */
  public static async overrideView(imodel: IModelConnection, viewState: ViewState) {
    const aspect = ViewSetup.getAspectRatio();
    if (undefined !== aspect) {
      viewState.adjustAspectRatio(aspect);
    }

    viewState.viewFlags = viewState.viewFlags.copy({ shadows: false, grid: false, visibleEdges: false });

    if (viewState.is3d()) {
      const viewState3d = viewState;
      const displayStyle = viewState3d.getDisplayStyle3d();

      displayStyle.changeBackgroundMapProps({ useDepthBuffer: true });

      const groundBias: number | undefined = await ViewSetup.getGroundBias(imodel);
      if (groundBias) {
        displayStyle.changeBackgroundMapProps({ groundBias });
```

**Comment by brinpat** [comment]
Another way I could resolve this is with the map layer widget addon, but I end up with an error when importing: import { MapLayersUI } from "@itwin/map-layers"; as @itwin has no map-layers dependency. I am unsure how to install map-layers from the npm site: https://www.npmjs.com/package/@itwin/map-layers

**Reply by mdastous-bentley** [reply]
Map-layers dependencies are listed as `peerDependencies` to avoid runtime conflicts, assuming your itwin.js application is already using it. Complete list available [here](https://github.com/iTwin/viewer-components-react/blob/01abc2da43943f43b5eb1afa6575c3b36d6edb22/packages/itwin/map-layers/package.json#L108).

Using the map-layers widget is the easiest way to have access to the different map-layers related settings.

**Reply by brinpat** [reply]
So at the moment my itwin.js has no map-layer in it. Would I need to add in the peer dependencies manually?

**Reply by mdastous-bentley** [reply]
Yes you need to install it... that said if you already have an itwin application, you probably have many of them already installed.

**Comment by mdastous-bentley** [comment]
You can invoke:
 `changeBackgroundMapProps({ terrainSettings: {heightOrigin: 1} });`

But keep this mind this is going to be a relative value, based on the active `TerrainProps,heightOriginMode`. Please read [documentation ](https://www.itwinjs.org/v2/reference/imodeljs-common/displaystyles/terrainheightoriginmode/) for more details about it.

**Reply by brinpat** [reply]
Thanks Michel, I will look into these and hopefully get the result I want soon

**Reply by brinpat** [reply]
I tried this out but nothing changed.

**Reply by mdastous-bentley** [reply]
Hi,

I created this [sandbox ](https://www.itwinjs.org/sandboxes/MichelDastous/elevation-correction) to demonstrate.

You can see, initially with only the terrain ON you get this:
![image](https://github.com/iTwin/community/assets/8309609/e98ee1f0-3a43-4b25-b3f5-a1259a408ff8)

Then if you set a value for `heightOrigin`: 
![image](https://github.com/iTwin/community/assets/8309609/57b8a4ea-e7d3-4bfb-9664-08485c58964f)

Also, the map-layers widget is installed in this sample, you can also use it to play with the settings.

HTH,

Michel

**Reply by brinpat** [reply]
Thanks Michel, I now believe there must be something overriding the height as that did not change anything either. I am looking into this

**Reply by brinpat** [reply]
Yes found it I believe GlobalDisplay widget was overwriting the terrain height

**Comment by brinpat** [comment]
```
viewport.changeBackgroundMapProps({applyTerrain: terrain});
      viewport.changeBackgroundMapProps({applyTerrain: true, terrainSettings: {heightOrigin: 100}});
```

Resolved by applying this code within Global Display widget
